### PR TITLE
Crash log test tools bug

### PR DESCRIPTION
**Pull request by evanchia-ly-sdets**
Fixes crash log name for the crash log watchdog.

Also addresses an issue for tests using editor utils where the crash logs were being renamed. Tested locally and all unit tests pass.

Signed-off-by: evanchia <evanchia@amazon.com>

### CODE REVIEW

**Review by Kadino — APPROVED** [review]
Looks good. There's an interesting amount of mock-changes this made necessary, but I'll defer any concerns about over-use of mocking.

File: Tools/LyTestTools/ly_test_tools/o3de/editor_test_utils.py [hunk 1]
```diff
@@ -71,7 +71,15 @@ def retrieve_crash_output(run_id: int, workspace: AbstractWorkspaceManager, time
     :return str: The contents of the editor crash file (error.log)
     """
     crash_info = "-- No crash log available --"
-    crash_log = os.path.join(retrieve_log_path(run_id, workspace), 'error.log')
+    error_log_regex = ""
+    log_path = retrieve_log_path(run_id, workspace)
+    # Gather all of the files in the log directory
+    dir_files = [f for f in os.listdir(log_path) if os.path.isfile(os.path.join(log_path, f))]
+    for file_name in dir_files:
+        # Search for all .log files with either "crash" or "error" because they could be renamed
```
Comment (AlexOteiza) [inline Tools/LyTestTools/ly_test_tools/o3de/editor_test_utils.py:79]: what does this mean with "renamed"?

File: Tools/LyTestTools/ly_test_tools/o3de/editor_test_utils.py [hunk 2]
```diff
@@ -71,7 +71,15 @@ def retrieve_crash_output(run_id: int, workspace: AbstractWorkspaceManager, time
...
+        if ("error" in file_name.lower() or "crash" in file_name.lower()) and (file_name.endswith(".log")):
+            crash_log = os.path.join(log_path, file_name)
+            break
```
Comment (AlexOteiza) [inline Tools/LyTestTools/ly_test_tools/o3de/editor_test_utils.py:82]: won't this find backuped crashes too?

File: Tools/LyTestTools/ly_test_tools/o3de/editor_test_utils.py (hunk 1 above)
Comment (evanchia-ly-sdets) [reply Tools/LyTestTools/ly_test_tools/o3de/editor_test_utils.py:79]: I'm not sure what is exactly renaming the log files, but we've seen error.log files being renamed i.e. error_2021_11_15_12_26_33.log

File: Tools/LyTestTools/ly_test_tools/o3de/editor_test_utils.py (hunk 2 above)
Comment (evanchia-ly-sdets) [reply Tools/LyTestTools/ly_test_tools/o3de/editor_test_utils.py:82]: I was under the assumption that the crash logs would overwrite themselves if another one existed

Comment (evanchia-ly-sdets) [reply Tools/LyTestTools/ly_test_tools/o3de/editor_test_utils.py:82]: Also I think that it should be the tool's or test's responsibility to clean up old crash logs

File: Tools/LyTestTools/ly_test_tools/o3de/editor_test_utils.py (hunk 1 above)
Comment (AlexOteiza) [reply Tools/LyTestTools/ly_test_tools/o3de/editor_test_utils.py:79]: those are backed up log files from previous runs. Don't search in these. The last ran test will always be error.log

Comment (evanchia-ly-sdets) [reply Tools/LyTestTools/ly_test_tools/o3de/editor_test_utils.py]: Thanks, found actual issue and submitted new changes

**Review by AlexOteiza — APPROVED** [review]
Looks good!

**Comment by AMZN-scspaldi** [comment]
Looks good!

**Review by AMZN-scspaldi — APPROVED** [review]

**Review by AMZN-Gene — APPROVED** [review]

**Pull request merged**